File: demo/ffi/ydraw/go/yplot/trunc.go
// Conceptual reproduction of demo/scripts/yplot/trunc.sh
// NOT RUNNABLE YET — conceptual, translated from the python sketch.
package main

import (
	"fmt"

	"github.com/zokrezyl/yetty/bindings/go/ydraw"
)

func show(plot ydraw.Plot) {
	dlist := ydraw.NewDrawableList()
	dlist.Add(plot)
	dlist.DcsEmit()
	dlist.Destroy()
}

func main() {

	// (1) The rounding family side by side — they split left of the origin.
	fmt.Println("(1) trunc vs floor vs round")
	show(ydraw.Plot{Width: 600, Height: 240, XRange: []float64{-3, 3}, YRange: []float64{-3.5, 3.5}, Functions: []ydraw.Function{
		{Body: "trunc(x)", Name: "truncated", Color: "#74C5A5"},
		{Body: "floor(x)", Name: "floored", Color: "#FF6B6B"},
		{Body: "round(x)", Name: "rounded", Color: "#FFE66D"},
	}})

	// (2) A symmetric sawtooth: x - trunc(x) keeps the sign of x.
	fmt.Println("(2) signed sawtooth vs fract")
	show(ydraw.Plot{Width: 600, Height: 240, XRange: []float64{-3, 3}, YRange: []float64{-1.1, 1.1}, Functions: []ydraw.Function{
		{Body: "x-trunc(x)", Name: "signed_saw", Color: "#6BA892"},
		{Body: "fract(x)", Name: "fract_saw", Color: "#556162"},
	}})

	// (3) A quantizer / ADC: a sine snapped onto discrete steps.
	fmt.Println("(3) quantizer")
	show(ydraw.Plot{Width: 600, Height: 240, XRange: []float64{0, 6.28}, YRange: []float64{-1.1, 1.1}, Functions: []ydraw.Function{
		{Body: "sin(x)", Name: "signal", Color: "#364A47"},
		{Body: "trunc(sin(x)*4)/4", Name: "quantized", Color: "#74C5A5"},
	}})

	// (4) A bit-crushed ramp: sample-and-hold staircase.
	fmt.Println("(4) bit-crushed ramp")
	show(ydraw.Plot{Width: 600, Height: 240, XRange: []float64{0, 4}, YRange: []float64{-0.2, 4.2}, Functions: []ydraw.Function{
		{Body: "x", Name: "ramp", Color: "#556162"},
		{Body: "trunc(x*3)/3", Name: "crushed", Color: "#6BA892"},
	}})
}

File: demo/ffi/ydraw/python/yplot/trunc.py
#!/usr/bin/env python3
"""Conceptual reproduction of demo/scripts/yplot/trunc.sh — NOT RUNNABLE YET."""
from yetty.ydraw import DrawableList, Function, Plot


def show(plot):
    dlist = DrawableList()
    dlist.add(plot)
    dlist.dcs_emit()
    dlist.destroy()


# (1) The rounding family side by side — they split left of the origin.
print('(1) trunc vs floor vs round')
show(Plot(width=600, height=240, x_range=(-3, 3), y_range=(-3.5, 3.5),
          functions=[
              Function("trunc(x)", name="truncated", color="#74C5A5"),
              Function("floor(x)", name="floored", color="#FF6B6B"),
              Function("round(x)", name="rounded", color="#FFE66D"),
          ]))

# (2) A symmetric sawtooth: x - trunc(x) keeps the sign of x.
print('(2) signed sawtooth vs fract')
show(Plot(width=600, height=240, x_range=(-3, 3), y_range=(-1.1, 1.1),
          functions=[
              Function("x-trunc(x)", name="signed_saw", color="#6BA892"),
              Function("fract(x)", name="fract_saw", color="#556162"),
          ]))

# (3) A quantizer / ADC: a sine snapped onto discrete steps.
print('(3) quantizer')
show(Plot(width=600, height=240, x_range=(0, 6.28), y_range=(-1.1, 1.1),
          functions=[
              Function("sin(x)", name="signal", color="#364A47"),
              Function("trunc(sin(x)*4)/4", name="quantized",
                       color="#74C5A5"),
          ]))

# (4) A bit-crushed ramp: sample-and-hold staircase.
print('(4) bit-crushed ramp')
show(Plot(width=600, height=240, x_range=(0, 4), y_range=(-0.2, 4.2),
          functions=[
              Function("x", name="ramp", color="#556162"),
              Function("trunc(x*3)/3", name="crushed", color="#6BA892"),
          ]))
